mitodl/ocw-studio
# Sync missing captions and transcripts

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #1518

#### What's this PR do?
Syncs captions and transcripts for missing videos
- Filter videos not having captions or transcripts
- Fetches respective file from 3play
- Create new WebsiteContent 
- Update existing video with new content object

#### How should this be manually tested?
- Make sure you have latest database dump locally
   - This is to ensure you have courses with missing captions/transcripts locally
- Make sure to add following .env variables: 
   - **Threeplay variables** 
      - `THREEPLAY_API_KEY`,  `THREEPLAY_CALLBACK_KEY`,  `THREEPLAY_PROJECT_ID`
   - **AWS variables**   
     - `AWS_STORAGE_BUCKET_NAME`,  `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`
-  Checkout to this branch
- **Testing all courses**
   - Run command `docker exec -it ocw-studio-web-1 python3 manage.py sync_management_command`
- **Testing single course**
   - Run command `docker exec -it ocw-studio-web-1 python3 manage.py sync_management_command --filter="name-of-the-course-year,"`
   - Add comma separated course names to be parsed
- You can stop the running command after a few minutes (Optional)
- Head to `http://localhost:8043/sites/`
- Search for the course name you provided
- Under Course Resources, and new transcripts/ captions resources should be present
- Open any resource, and you should see captions/transcripts file paths (previously empty)


### Example of captions/ transcript conent objects and paths
<img width="1338" alt="Screenshot 2023-04-06 at 5 09 33 PM" src="https://user-images.githubusercontent.com/68846543/230375034-9e0eca4f-5a31-42ea-b55b-cb9a3922f48c.png">
<img width="1409" alt="Screenshot 2023-04-06 at 5 09 46 PM" src="https://user-images.githubusercontent.com/68846543/230375052-4bf9a032-e1bd-4ff9-a5c0-4fb604bf0a0c.png">
